core/src/main/java/com/amplifyframework/datastore/events/SyncQueriesStartedEvent.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.events;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.Arrays;
+
+/**
+ * Event payload emitted when the sync process starts.
+ */
+public final class SyncQueriesStartedEvent {
+    private final String[] models;
```

Comment:
Perhaps we can use `List<String>` or `Set<String>` here?

Most de/serializers should be able to easily convert these collections into a json array.

Reply:
I'll probably do this as a follow up PR.